doc/index.rst

```diff
@@ -6,7 +6,100 @@
+functionality from a group of components. Users can use these classes
+separately to control a set of components or combine them to achieve even
+higher levels of operations.
+
+For instance, the `ATCS` class combine functionality for all
+"telescope-related" components of the Vera Rubin Auxiliary Telescope. Actions
+like slewing the telescope, while making sure all involved components are in
+`ENABLED` state and wait until all components are in position is as simple as;
+
+::
+
+   from lsst.ts.observatory.control import ATCS
+
+   atcs = ATCS()
+
+   await atcs.start_task
+
+   await atcs.slew_object(name="Alf Pav")
+
+A set of utilities are also provided that allow users to combine component
+coordination with activities like data analysis and more.
+
+
+The repo defines the `RemoteGroup` class which
+represent a group of components in the control system. This class implements
+basic functionality that is common to all groups of CSCs.
+
+Limiting required resources
+===========================
+
+In some cases, the high level class may need to communicate with several
+components at the same time to achieve high-level operations. Some of these
+components can have high frequency topics which will make the remotes quite
+heavy to operate. At the same time, depending on the number of remotes a group
+defines, or the number of groups needed by a certain operation, the overall
+resources might also be quite heavy. To work around this the `RemoteGroup` base
+class provides ways for users to limit the resources to the minimum set
+required for the intended usage. This is controlled by the optional parameter
+`intended_usage` in `RemoteGroup`.
+
+By default `intended_usage = None`, which causes the class to load all the
+resources from the remotes. This can be very helpful when using Jupyter
+notebooks or testing environments where the user need more flexibility.
+
+The `Usages` class contains the usages definition. By default, only a small
+subset of generic operations are defined, e.g., `All`, `StateTransition`,
+`MonitorState` and `MonitorHeartBeat`. Note that the meaning of `Usages.All`
+is to load the resources for all the defined operations and not to load all the
+resources.
+
+The resources needed for each operation are defined in the `usages` property of
+the `RemoteGroup` class. This class property defines a dictionary where the
+keys are the usages and the values are the resources. The resources are
+instances of `types.SimpleNamespace` with attributes;
+
+`components`
+    List of components required for the usage. If a component is part of a
+    group but it is not needed for a certain type of usage, the component will
+    not be instantiated during the initialization procedure.
+
+`readonly`
+    Defines an operation as read only. This will cause all commands to be
+    ignored.
+
+`include`
+    List with topics to include in the operation. The list is common to all
+    components to avoid duplication of generic topics.
```

Comment:
You may find at some point that you want to add `exclude` as well. salobj supports this and it is a useful way of getting rid of one or a few chatty topics (e.g. M1M3 telemetry) while allowing the rest.